src/linear_algebra/eigenspace.lean

```diff
@@ -0,0 +1,242 @@
+/-
+Copyright (c) 2020 Alexander Bentkamp. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Alexander Bentkamp.
+-/
+
+import field_theory.algebraic_closure
+import linear_algebra.finsupp
+
+/-!
+# Eigenvectors and eigenvalues
+
+This file defines eigenspaces and eigenvalues.
+
+An eigenspace of a linear map `f` for a scalar `μ` is the kernel of the map `(f - μ • id)`.
+The nonzero elements of an eigenspace are eigenvectors `x`. They have the property `f x = μ • x`.
+If there are eigenvectors for a scalar `μ`, the scalar `μ` is called an eigenvalue.
+
+## Notations
+
+The expression `algebra_map α (β →ₗ[α] β)` appears very often, which is why we
+use `am` as a local notation for it.
+
+## References
+
+* [Sheldon Axler, *Down with determinants!*,
+  https://www.maa.org/sites/default/files/pdf/awards/Axler-Ford-1996.pdf][axler1996]
+* https://en.wikipedia.org/wiki/Eigenvalues_and_eigenvectors
+
+## Tags
+
+eigenspace, eigenvector, eigenvalue, eigen
+-/
+
+universes u v w
+
+open vector_space principal_ideal_ring polynomial finite_dimensional
+
+variables {α : Type v} {β : Type w} [add_comm_group β]
```

Comment:
It will be an annoying change, but could you replace `α` with `K` and `β` with `V` everywhere? I prefer the mathematical convention of indicating the structure on a variable using the variable's name. They seem to be consistently used for a field and a vector space respectively.

Reply:
Not annoying at all. Search and replace did it without problem.